### PR TITLE
RPRBLND-2301: Fix addon enabling in Blender 4.0+

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Fix addon enabling in Blender 4.0+

### EFFECT OF CHANGE
Initial support for Blender 4.0+

### TECHNICAL STEPS
Adjusted node categories registration.
Adjusted `ShaderNodeBsdfPrincipled`.
Refactored sockets from `NodeSocketInterface`  to `NodeTreeInterfaceSocket`.
Migration from `bgl` to `gpu` module.
Fixed render passes dimensions issue.

### NOTES FOR REVIEWERS
Removed `GL Interoperability` option due to migration to `gpu` module.

### CODE REVIEW

File: src/rprblender/engine/viewport_engine.py [hunk 1]
```diff
@@ -745,107 +741,40 @@ def _sync_update_before(self):
...
-
             if self.width * self.height == 0:
                 return
 
             im = self._get_render_image()
 
         self.gl_texture.set_image(im)
-        self.draw_texture(self.gl_texture.texture_id, scene)
+        self.draw_texture(scene)
+        self.gl_texture.clear()
```
Comment (bnagirniak) [inline src/rprblender/engine/viewport_engine.py:777]: do we need to call `clear()`?

File: src/rprblender/nodes/__init__.py [hunk 2]
```diff
@@ -188,10 +187,31 @@ def func(cls, context):
...
+    if bpy.context.scene.render.engine != "RPR":
+        return
+
+    for node in nodes:
+        op = col.operator("node.add_node", text=node.bl_label)
+        op.type = node.__name__
+        op.use_transform = True
+
+
+RPR_NODES = {'shader': (rpr_nodes.RPRShaderNodeUber,
```
Comment (bnagirniak) [inline src/rprblender/nodes/__init__.py:204]: assign this to node_categories:
```
if BLENDER_VERSION >= "4.0":
    node_categories = {...}
else:
    node_categories = [...]
```

File: src/rprblender/engine/render_engine.py [hunk 3]
```diff
@@ -150,10 +148,7 @@ def set_render_result(render_passes: bpy.types.RenderPasses):
 
                     self.cached_rendered_images[p.name][y1:y2, x1:x2] = image
 
-                images.append(image.flatten())
-
-            # efficient way to copy all AOV images
-            render_passes.foreach_set('rect', np.concatenate(images))
+                p.rect = image.reshape(len(p.rect), -1)
```
Comment (bnagirniak) [inline src/rprblender/engine/render_engine.py:151]: `foreach_set` should be used

**Review by bnagirniak — CHANGES_REQUESTED** [review]

File: src/rprblender/utils/gl.py [hunk 4]
```diff
@@ -12,9 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #********************************************************************
-import platform
 import numpy as np
-import bgl
+import gpu
 
 import ctypes
 import sys
```
Comment (bnagirniak) [inline src/rprblender/utils/gl.py:19]: Remove `gl` module load

**Review by bnagirniak — CHANGES_REQUESTED** [review]

**Review by bnagirniak — APPROVED** [review]
Tested - works good, Added small code improvements.

**Review by ARozghon — APPROVED** [review]
Approved by QA

**Pull request merged**